armi/nuclearDataIO/rtflux.py

```diff
@@ -24,180 +41,172 @@
 import matplotlib.patches as mpatches
 from matplotlib.collections import PatchCollection
 
-from armi.nuclearDataIO import cccc
+from armi.nuclearDataIO import cccc, nuclearFileMetadata
 from armi.reactor import locations
 
 
-class RTFLUX(cccc.CCCCReader):
-    """
-    Read a binary RTFLUX or ATFLUX file from DIF3D output.
-
-    While NHFLUX stores nodal flux output data, RTFLUX stores finite difference flux output data.
-    While NHFLUX stores data by hexagon, RTFLUX stores data by triangle.
-    This can also read ATFLUX, which stores the finite difference adjoint flux output data.
-    """
-
-    def __init__(self, fName="RTFLUX"):
-        r"""
-        Initialize the RTFLUX or ATFLUX reader object.
-
-        Parameters
-        ----------
-        fName : str, optional
-            The file name of the RTFLUX/ATFLUX binary file to be read.
-
-        """
-        cccc.CCCCReader.__init__(self, fName)
+# See CCCC-IV documentation for definitions
+FILE_SPEC_1D_KEYS = (
+    "NDIM",
+    "NGROUP",
+    "NINTI",
+    "NINTJ",
+    "NINTK",
+    "ITER",
+    "EFFK",
+    "POWER",
+    "NBLOK",
+)
 
-        self.fc = {}  # file control info (sort of global for this library)
-        self.triangleFluxes = []
 
-        self.readFileID()  # Read RTFLUX file ID
-
-    def readAllData(self):
-        r"""
-        Read multigroup fluxes from the original DIF3D FD triangular-z mesh points.
-        RTFLUX contains the real fluxes, while ATFLUX contains the adjoint fluxes.
-
-        """
-
-        # Read basic data parameters (number of energy groups, assemblies, axial nodes, etc.)
-        self.read1D()
+class RtfluxData:
+    """
+    Data structure that is read from or written to a RTFLUX file.
```

Comment:
Its really nice to have separated this out from the act of reading, but now the question is... what is it?